Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp

```diff
@@ -1116,6 +1115,53 @@ namespace AZ::Data
         return asset;
     }
 
+    void AssetManager::QueueAssetReload(AZ::Data::Asset<AZ::Data::AssetData> newAsset, bool signalLoaded)
+    {
+        AssetHandler* handler = nullptr;
+
+        // Resolve the asset handler and allocate new data for the reload.
+        {
+            AssetHandlerMap::iterator handlerIt = m_handlers.find(newAsset.GetType());
+            AZ_Assert(
+                handlerIt != m_handlers.end(), "No handler was registered for this asset [type:%s id:%s]!",
+                newAsset.GetType().ToString<AZ::OSString>().c_str(), newAsset.GetId().ToString<AZ::OSString>().c_str());
+            handler = handlerIt->second;
+        }
+
+        AZStd::shared_ptr<AssetDataStream> dataStream;
+        AssetStreamInfo loadInfo = GetModifiedLoadStreamInfoForAsset(newAsset, handler);
+        constexpr bool isReload = true;
+        if (loadInfo.IsValid())
+        {
+            // Create the AssetDataStream instance here so it can claim an asset reference inside the lock (for a total
+            // count of 2 before starting the load), otherwise the refcount will be 1, and the load could be canceled
+            // before it is started, which creates state consistency issues.
+
+            dataStream = AZStd::make_shared<AssetDataStream>(handler->GetAssetBufferAllocator());
+            if (dataStream)
+            {
+                // Currently there isn't a clear use case for needing to adjust priority for reloads so the default load priority is used
+                QueueAsyncStreamLoad(newAsset, dataStream, loadInfo, isReload, handler, {}, signalLoaded);
+            }
+            else
+            {
+                AZ_Assert(
+                    false, "Failed to create dataStream to reload asset %s (%s)", newAsset.GetId().ToString<AZ::OSString>().c_str(),
+                    newAsset.GetHint().c_str());
+            }
+        }
+        else
+        {
+            // Asset creation was successful, but asset loading isn't, so trigger the OnAssetError notification
+            AZ_Error(
+                "AssetDatabase", false, "Failed to retrieve required information for asset %s (%s)",
```

Comment:
It looks like this would either fail because there is no handler, which will print out its own error message, or because the handler chose to return nothing, which isn't something we'd have any info about why.  I'll update it a little bit but there's not much for info I can add.  This is also existing code, I've just moved it around.